GDJS/Runtime/howler-sound-manager/howler-sound-manager.js

```diff
@@ -68,65 +54,35 @@ gdjs.HowlerSound = class HowlerSound {
    * @returns {HowlerSound} The current instance for chaining.
    */
   play() {
-    this._id = this._howl.play(this._id === null ? undefined : this._id);
-
-    if (this._id === null) {
-      this._paused = true;
-      this._stopped = true;
-
+    if (this._howl._state === 'loaded') {
+      this._id = this._howl.play(this._id === null ? undefined : this._id);
+    } else {
+      this._loading = true;
       // In case the howl didn't finsih loading, start the sound once it finishes loading.
       this._howl.once('load', () => {
-        this._hasListeners = false;
+        this._loading = false;
         this.play();
       });
       // Make sure it is loading
       this._howl.load();
-      // Directly return to not create unecessary listeners
-      return this;
     }
 
-    this._paused = false;
-    this._stopped = false;
-
-    if (!this._hasListeners) {
-      this._hasListeners = true;
-      /*
-       * Track events to be sure the status is
-       * sync'ed with the sound - though some should be redundant
-       * with `play`/`pause` methods already doing that. Keeping
-       * that to be sure that the status is always correct.
-       *
-       * This is done in play as we need the ID to set up those callbacks.
-       */
-
-      this.on('play', () => {
-        this._paused = false;
-        this._stopped = false;
-      });
-
-      this.on('pause', () => {
-        this._paused = true;
-        this._stopped = false;
-      });
-
-      this.on('end', () => {
-        if (!this._howl.loop()) {
-          this._canBeDestroyed = true;
-          this._paused = false;
-          this._stopped = true;
-        }
-      });
-
-      this.on('playerror', (id, error) => {
-        console.error(
-          "Can't play a sound, considering it as stopped. Error is: ",
-          error
-        );
-        this._paused = false;
-        this._stopped = true;
-      });
-    }
+    return this;
+  }
 
+  /**
+   * Sets the current sound as a free sound.
+   * This will set up a listener to mark the sound as destroyable
+   * after it ended.
+   * This has to be done separately as it could lead to a memory leak otherwise.
+   * @returns {HowlerSound} The current instance for chaining.
+   */
+  _setFree() {
```

Comment:
Technically, this should be redundant and the store in array method should be able to check if the sound is stopped to know if it can be replaced.

Reply:
True, though maybe in the past this was necessary because just checking if the sound is stopped would return `true` if the sound was loading, so it would delete a sound from the free sounds array while it was not even launched?

I agree this kind of seems redundant now, but I would be worried of such race conditions.

If we keep this, do you think we could also do this for musics/sounds on channels? I know that we're not using `canBeDestroyed` on them, but for consistency I think it's better to avoid doing any difference between a free and a channel sound/music. So we could set up this listener in all cases? what do you think?